service/history/workflow/mutable_state_impl.go

```diff
@@ -4265,7 +4265,10 @@ func (ms *MutableStateImpl) ApplyWorkflowExecutionUpdateAcceptedEvent(
 	if ms.executionInfo.UpdateInfos == nil {
 		ms.executionInfo.UpdateInfos = make(map[string]*persistencespb.UpdateInfo, 1)
 	}
-	updateID := attrs.GetAcceptedRequest().GetMeta().GetUpdateId()
+	// NOTE: `attrs.GetAcceptedRequest().GetMeta().GetUpdateId()` was used here before, but that is problematic
+	// in a reset/conflict resolution scenario where there is no `acceptedRequest` since the previously written
+	// UpdateAdmitted event already contains the Update payload.
```

Comment:
I would rephrase it to:
1. not refer "previously" case,
2. support future durable admitted case.

```suggestion
	// NOTE: `attrs.GetAcceptedRequest().GetMeta().GetUpdateId()` can't be used here,
	// because `acceptedRequest` might be nil if UpdateAdmitted event with the Update payload
	// was previously written to the history. 
	// For example, in a reset/conflict resolution scenario.
```